plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -63,25 +88,22 @@ export class TeamManager {
 
         if (!this.eventNamesCache.get(team.id)?.has(event)) {
             await this.db.postgresQuery(
-                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id, last_seen_at, created_at) VALUES ($1, $2, NULL, NULL, $3, $4, NOW()) ON CONFLICT DO NOTHING`,
+                `INSERT INTO posthog_eventdefinition (id, name, volume_30_day, query_usage_30_day, team_id, last_seen_at, created_at)` +
+                    ` VALUES ($1, $2, NULL, NULL, $3, $4, NOW())` +
+                    ` ON CONFLICT ON CONSTRAINT posthog_eventdefinition_team_id_name_80fa0b87_uniq` +
+                    ` DO UPDATE SET last_seen_at=$4`,
                 [new UUIDT().toString(), event, team.id, eventTimestamp],
                 'insertEventDefinition'
             )
             this.eventNamesCache.get(team.id)?.add(event)
-            this.eventLastSeenCache.set(`${team.id}_${event}`, eventTimestamp)
+            this.eventLastSeenCache.set(`${team.id}_${event}`, eventTimestamp.valueOf())
         } else {
-            if (
-                !this.eventLastSeenCache.get(`${team.id}_${event}`) ||
-                eventTimestamp.diff(
-                    this.eventLastSeenCache.get(`${team.id}_${event}`) || DateTime.fromISO('1970-01-01')
-                ).minutes > 15
-            ) {
-                await this.db.postgresQuery(
-                    `UPDATE posthog_eventdefinition SET last_seen_at = GREATEST(last_seen_at, $1) WHERE name = $2 AND team_id = $3`,
-                    [eventTimestamp.toUTC().toISO(), event, team.id],
-                    'updateEventLastSeen'
-                )
-                this.eventLastSeenCache.set(`${team.id}_${event}`, eventTimestamp)
+            if ((this.eventLastSeenCache.get(`${team.id}_${event}`) ?? 0) < DateTime.now().valueOf()) {
+                this.eventLastSeenCache.set(`${team.id}_${event}`, eventTimestamp.valueOf())
+            }
+            if (this.eventLastSeenCache.size > 100000 || DateTime.now().diff(this.lastFlushAt).minutes > 60) {
```

Comment:
Q: Why did we end up going with this approach? IIRC everyone was pushing towards using a cronjob if possible.